src/fec_context.h

```diff
@@ -40,13 +40,27 @@
 
 #include "fft_base.h"
 #include "gf_base.h"
+#include "gf_nf4.h"
 #include "vec_poly.h"
 #include "vec_zero_ext.h"
 
 namespace nttec {
 
 namespace fec {
 
+/* List of vectors handled by context */
+enum vector_type {
```

Comment:
use `enum class`: they are strongly typed and the enumerators are scoped (unlike C-like enum).
Since they are scope, you could write:

```C++
enum class CtxVec {
    A = 0,
    A_FFT_2K,
    INV_A_I,
    S,
    N1,
    N2,
    2K_1,
    2K_2
};
```
And then use it as `CtxVec::A`, `CtxVec::2K_1`, …

Also, do you need `vec_end`? Seems unused to me.